src/zenml/config/step_configurations.py

```diff
@@ -146,6 +146,27 @@ class StepConfigurationUpdate(StrictBaseModel):
         "name"
     )
 
+    @root_validator
+    def _disable_cache_if_new_model_version_requested(
```

Comment:
But I will leave this up to you

Reply:
I think I'd expect the new model version to be created as implemented. Safer than the opposite expectation, too, I think.

Reply:
I added `update_new_versions_requests` to ensure that model configurations are enriched with new model version requests, if any of the steps or pipeline requested it for a given model name. And I removed the disabled cache validator, which makes more sense to me.
There is a test demonstrating this behavior. 
Can you closely check https://github.com/zenml-io/zenml/pull/1847/commits/99abf715176170862d6d1e7256f073a666d2c6e2, @strickvl @fa9r ?

Reply:
On top: we got feedback that the creation of a new model version with a given name might also come in handy. I'll look into this in separate PR later on - current implementation relies solely on the assumption that there can be only one running model version. Generally, it might be not the best idea to assume so, due to parallel execution cases (e.g. search for best model configuration in multiple runs in one shot).

Reply:
I'll proceed to merge this, feel free to comment in mega-branch PR if something is off here.